src/llm_vm/onsite_llm.py

```diff
@@ -173,6 +196,35 @@ def generate(self,prompt,max_length=100,**kwargs): # both tokenizer and model ta
         resp= self.tokenizer.batch_decode(generate_ids,skip_special_tokens=True,clean_up_tokenization_spaces=False)[0]
         # need to drop the len(prompt) prefix with these sequences generally 
         return resp[len(prompt):]
+    
+    def finetune(self,data, optimizer, c_id):
+        old_model = optimizer.storage.get_model(c_id)
+        final_dataset = []
+        for i in data:
+            final_dataset =  i[0] + i[1]
+        f_dataset = map(self.tokenizer,final_dataset)
+        self.tokenizer.pad_token = self.tokenizer.eos_token
+        data_collator = DataCollatorForLanguageModeling(tokenizer=self.tokenizer, mlm=False)
+
+        training_args = TrainingArguments(
+            output_dir="Neo_finetuned",
+            evaluation_strategy="epoch",
+            learning_rate=2e-5,
+            num_train_epochs=2,
+            weight_decay=0.01,
+        )
+
+        trainer = Trainer(
+            model=self.model,
+            args=training_args,
+            train_dataset=FinetuningDataset(f_dataset,len(final_dataset)),
+            eval_dataset=FinetuningDataset(f_dataset,len(final_dataset)),
+            data_collator=data_collator,
+        )
+
+        trainer.train()
```

Comment:
Does this update the model in place ?

Reply:
no creates a weights file that gets saved